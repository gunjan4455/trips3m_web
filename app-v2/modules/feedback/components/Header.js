import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { CloseIconWhite } from 'helpers/Icon/Icon';

class Header extends Component {
  static propTypes = {
    text: PropTypes.string
  };

  render() {
    return (
      <div className='pbc1 flex alignCenter pl15 pr15 pt8 pb8 fixed wfull'>
        <div className='pt4 pb4 flex alignCenter'>
          <Link to={'/'}>
            <span className='input-24 block mr15'><CloseIconWhite /></span>
          </Link>
          <p className='f16 fw7 m0 sfcw'>{this.props.text}</p>
        </div>
      </div>
    );
  }
}

export default Header;
