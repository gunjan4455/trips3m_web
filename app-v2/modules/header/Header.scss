// app-shell css part
.tt-logo-container {
  width: 140px;
  height: 18px;
  position: relative;
  & svg {
    width: 140px;
    height: 18px;
  }
}
header {
  position: relative;
  box-shadow: none;
}
.side-nav-con {
  padding: 0;
  background: #FFFFFF;
  width: 280px;
}
.nav-con-links {
  list-style: none;
  padding: 24px 0;
  margin-bottom: 16px;
  border-bottom: 1px solid #D4D4D4;
  li {
    display: block;
  }
  a {
    color:#2B2B2B;
    text-decoration: none;
    padding: 8px 16px;
    display: block;
  }
  a:hover{
    color:#2B2B2B;
    text-decoration: none;
  }
}
.header-con {
  background: #20A397;
  padding: 12px 10px;
  position: relative;
}
.header-con-transparent {
  background: transparent;
  padding: 12px 10px;
  position: relative;
}
.header-con-shadow {
  box-shadow: none;
}
.header-con-first {
  height: 22px;
  padding-left: 10px;
  padding-right: 10px;
}
.burger-menu {
  width: 20px;
  height: 16px;
  float:left;
  padding-top: 2px;
}
.burger-menu .sidenav-link{
  width: 20px;
  height: 14px;
  display: block;
  border-top: 1px solid #FFFFFF;
  border-bottom: 1px solid #FFFFFF;
  position: relative;
}
.burger-menu .sidenav-link::before {
  height: 0;
  margin-top: -1px;
  border-bottom: 1px solid #FFFFFF;
  width: 100%;
  position: absolute;
  top: 50%;
  left: 0;
  content: '';
}
.tt-logo {
  width: 140px;
  float:left;
  margin-left: 16px;
}
.add-to-screen {
  width: 14px;
  height: 20px;
}
.add-to-home {
  border: 1px solid #FFFFFF;
  height: 20px;
  line-height: 18px;
  width: 14px;
  display: block;
  border-radius: 2px;
  color: #FFFFFF;
  text-decoration: none;
  text-align: center;
}
.add-to-home::before {
  content: '+';
}
.add-to-home-button {
  display: none;
}
.add-to-home-button + label {
  display: block;
  float: right;
  width: 14px;
  height: 20px;
}
.add-to-home-info {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 12;
  display: none;
  overflow: hidden;
}
.add-to-home-button:checked + label .add-to-home-info{
  display: block;
}
.addtohomeoverlay{
  background-color:#20A397;
    width: 32px;
    height: 32px;
    border-radius: 100px;
    text-align:center;
    position: relative;
    top: 10px;
}
.search-header-large {
  height: 48px;
  background: #FFFFFF;
  font-size: 14px;
  width: 100%;
  border-radius: 2px;
  padding: 16px 50px 16px 16px;
  border: 0 none;
}
.search-header {border-radius: 2px; position: relative; margin-top: 0px; height: 48px;}
.search-header::after {
  content:'';
  width: 15px;
  height: 15px;
  border-radius: 100%;
  position: absolute;
  right: 24px;
  top: 16px;
  border:1px solid #7F7F7F;
}
.search-header::before {
  content: '';
  width: 1px;
  height: 10px;
  position: absolute;
  right: 22px;
  background: #7F7F7F;
  transform: rotate(-44deg);
  top: 28px;
}
.search-header-common {
  position: relative;
  width: 22px;
  height: 22px;
  float: right;
  background-color: transparent;
  border: 0 none;
  padding-top: 22px;
  overflow: hidden;
}
.search-header-common::after {
  content: '';
      width: 15px;
      height: 15px;
      border-radius: 100%;
      position: absolute;
      right: 6PX;
      top: 0;
      border: 1px solid #FFFFFF;
}
.search-header-common::before {
  content: '';
width: 1px;
height: 10px;
position: absolute;
right: 5px;
background: #FFFFFF;
transform: rotate(-44deg);
top: 12px;
}
.header-title {
  background: #00786c;
  padding: 12px 15px;
}
.header-title-p {
  display: flex;
}
.back-arrow {
width: 30px;
min-width: 20px;
height: 20px;
float: left;
display:  block;
margin-right: 2px;
position: relative;
padding: 10px;
text-indent: -100px;
overflow: hidden;
background-color: transparent;
border: 0 none;
}
.back-arrow::before {
content: '';
width: 10px;
height: 1px;
background: #fff;
transform: rotate(-45deg);
display: block;
position: absolute;
top: 6px;
left: 5px;
}
.back-arrow::after {
content: '';
  display: block;
  width: 10px;
  height: 1px;
  background: #fff;
  transform: rotate(45deg);
  position: absolute;
  top: 13px;
  left: 5px;
}
.burger-menu-input {
  display: none;
}
.burger-menu-input + label {
  display: block;
}
.sidenav-container {
  width: 280px;
  background: #ffffff;
  height: 100%;
  left: 0;
  top: 0;
  bottom: 0;
  z-index: 21;
  position: fixed;
  display: none;
  overflow: auto;
}
.sidenav-blacklayer {
  z-index: 20;
  position: fixed;
  background: rgba(0,0,0,0.6);
  width: 100%;
  height: 100%;
  left:0;
  top: 0;
  display: none;
}
.burger-menu-input:checked + label .sidenav-container , .burger-menu-input:checked + label .sidenav-blacklayer{
  display: block;
}
.tap-tag-addtohome{line-height:24px;
padding-right:6px;}
// app-shell css ends here
.down-arrow-destination {
  width: 12px;
  height: 6px;
  position: relative;
  margin-left: 20px;
  display: inline-block;
    &:before, &:after {
      content: '';
      position: absolute;
      background: #FFFFFF;
      height: 1px;
      width: 7px;
      top: 2px;
    }
    &:before {
      transform: rotate(45deg);
      left: 0;
    }
    &:after {
      transform: rotate(135deg);
      right: 0;
    }
}
.category-list {
  left: 0;
  top: 90px;
  z-index: 11;
  ul {
    margin:0;
    padding: 0;
    z-index: 2;
    position: relative;
    background: #FFFFFF;
    li {
      list-style: none;
      a {
        color: #2B2B2B;
        display: block;
        padding: 15px;
        &:hover, &:active, &:visited, &:focus {
          color: #2B2B2B;
        }
        &.selected-category {
          color: #FE5246;
        }
      }
    }
  }
  .fix-layer-suggestion {
    width: 100%;
    height: 100%;
    background: rgba(0,0,0,0.4);
    position: fixed;
    top: 90px;
    left: 0;
    z-index: 1;
  }
}
.button-drop-category {
  button {
    background: transparent;
    border: 0 none;
    color: #FFFFFF;
    font-size: 14px;
    text-align: left;
    min-width: 180px;
  }
}
.transparent-cut {
  position: absolute;
  width: 70px;
  height: 70px;
  top: -2015px;
  right: -2015px;
  background: transparent;
  border-radius: 100%;
  box-sizing: content-box;
  border: 2000px solid #000;
  opacity: 0.95;
}
.header-title .share-logo-right a {
  height: 25px;
  width: 25px;
  display: block;
}

.homePage .header-con-transparent {
  padding: 16px 10px;
}
.homePage .tt-logo {
  margin-left: 0;
}
