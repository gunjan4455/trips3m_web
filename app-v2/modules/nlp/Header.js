import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import BudgetRange from './BudgetRange';

import styles from './Preferernces.cm.scss';


class Header extends PureComponent {
  render() {
    const { price } = this.props;
    return (
      <div>
        <div className="at_nurturingBanner relative sbcw pb32">
          <div className={`at_nurturingDiv pbc5 pt15 ${styles.bannerHeading}`}>
            <p className="f16a sfcw fw9 pl48">Edit Preferences to Match Your Budget</p>
          </div>
          <div className={`absolute wfull ${styles.budgetWrap}`}>
            <BudgetRange priceMax={price.max} priceMin={price.min} />
          </div>
        </div>
      </div>
    );
  }
}

Header.propTypes = {
  price: PropTypes.object
};

Header.defaultProps = {
  price: {}
};

export default Header;
