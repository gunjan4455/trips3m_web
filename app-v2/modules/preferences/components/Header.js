import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Back2IconWhite } from 'helpers/Icon/Icon';
import { rtripPath } from 'helpers/urlHelpers';

class Header extends Component {
  static propTypes = {
    currentTripId: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string
    ]),
    goBack: PropTypes.func.isRequired
  };

  render() {

    return (
      <div className='pbc1 flex alignCenter pl15 pr15 pt8 pb8'>
        <div className='pt4 pb4 flex'>
          <span className='input-24 block mr15' onClick={this.props.goBack}><Back2IconWhite /></span>
          <p className='f16 fw7 m0 sfcw'>{`Booking ID #${this.props.currentTripId}`}</p>
        </div>
      </div>
    );
  }
}

export default Header;
