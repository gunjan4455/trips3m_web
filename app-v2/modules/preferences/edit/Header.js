import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Header extends Component {
  static propTypes = {
    text: PropTypes.string
  };

  static defaultProps = {
    text: 'Edit Preference'
  };

  render() {
    return (
      <div className="pt8 pb8 pl15 pr15 flex bb wfull fixed z5 sbcw">
        <div className="pt4 pb4 flex">
          <p className="f16 flex alignCenter fw7 flexFull ellipsis ml24">
            {this.props.text}
          </p>
        </div>
      </div>
    );
  }
}

export default Header;
