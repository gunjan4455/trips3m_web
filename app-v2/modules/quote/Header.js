import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { Back2IconWhite } from 'helpers/Icon/Icon';
import { rtripPath, downloadQuoteUrl } from 'helpers/urlHelpers';
import Notification from 'modules/notifications';
import { getFullName } from 'helpers/utils';
import SocialShare from 'modules/quote/ctas/SocialShare';
import { formatURL } from 'utils/parsers';
import { trackSegment } from 'actions/segmentEvents';

class Header extends Component {
  static propTypes = {
    tripId: PropTypes.number.isRequired,
    trackSegment: PropTypes.func,
    currentQuoteId: PropTypes.number.isRequired,
    userDetails: PropTypes.object.isRequired
  };

  static defaultProps = {
    trackSegment: () => {}
  };

  trackSegmentEvent = (event, cta, object = '') => {
    this.props.trackSegment({
      event,
      section: 'Sticky Bar',
      object: object,
      cta
    });
  };

  render() {
    const { currentQuoteId, userDetails } = this.props;
    const NotificationContainer = Notification.container;
    const shareLink = formatURL({ path: downloadQuoteUrl(currentQuoteId) });

    return (
      <div className="pbc1 p8 flex">
        <a
          href={rtripPath(this.props.tripId)}
          className="mr8 input-24 p4 iblock"
        >
          <Back2IconWhite />
        </a>
        <p className="f16 flex alignCenter fw7 sfcw flexFull ellipsis">
          Quote Details
        </p>
        <SocialShare
          sharedBy={getFullName(userDetails.first_name, userDetails.last_name)}
          shareLink={shareLink} key="cta-social-share"
          trackSegment={this.trackSegmentEvent}
        />
        <NotificationContainer />
      </div>
    );
  }
}

export default Header;
