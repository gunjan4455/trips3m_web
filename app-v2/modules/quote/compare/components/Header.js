import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Back2IconWhite } from 'helpers/Icon/Icon';

class Header extends Component {
  static propTypes = {
    goBack: PropTypes.func.isRequired,
  };

  render() {
    return (
      <div className="pbc1 p8">
        <div className="p4 flex">
          <a
            onClick={this.props.goBack}
            className="mr8 input-24 p4 iblock at_qcp_backBtn"
          >
            <Back2IconWhite />
          </a>
          <p className="f16 flex alignCenter fw7 sfcw flexFull ellipsis at_qcp_headingTop">
            Compare Quotes
          </p>
        </div>
      </div>
    );
  }
}

export default Header;
