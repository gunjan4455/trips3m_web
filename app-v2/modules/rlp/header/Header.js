import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

import { TRIP_STATUS } from 'constants/tripStatus';
import { CompareIcon, CompareIconWhite } from 'helpers/Icon/Icon';
import { compareQuotesPath } from 'helpers/urlHelpers';
import TripList from 'modules/trip/dropdown/index';
import Modal from 'modules/shared/GenericModal';
import CancelRequest from 'modules/trip/cancelRequest';
import ViewPreferencesButton from 'modules/shared/ViewPreferencesButton';
import { isTripConverted, rescue } from 'helpers/utils';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';

const GViewPreferencesBtn = styled.div({
  flex: '0 0 135px',
});

class Header extends Component {

  constructor(props) {
    super(props);

    this.state = {
      closeModal: true
    };
  }

  reactivateBooking = () => {
    const { reactivateBooking, currentTrip } = this.props;
    reactivateBooking({ tripId: currentTrip.id }).then((results) => {
      if (results) {
        window.location.reload();
      }
    });
  };

  trackSegmentEvents = () => {
    rescue(() => {
      this.props.trackSegment({
        event: ORGANISM_CLICKED,
        section: 'Trip request details',
        object: '',
        cta: 'Cancel Request'
      });
    });
  };

  closePopup = () => {
    this.setState({ closeModal: true });
  };

  isCancelledTrip = (status) => status.toLowerCase() === TRIP_STATUS.CANCELLED;


  renderCancel() {
    const { currentTrip } = this.props;

    return this.isCancelledTrip(currentTrip.status) ?
      <button className='sfc1 fw7 f14 at_undoCancel_cta' onClick={this.reactivateBooking}>Undo Cancel</button> :
      <Modal fullView={true} hasCustomClose={true}
             trigger={<span className='f14 fw7 sfc6 at_cancelRequest' onClick={this.trackSegmentEvents}>Cancel Request</span>}
             customCloseTrigger={this.state.closeModal}>
        <div className='sbcw'>
          <CancelRequest closeModal={this.closePopup} tripId={currentTrip.id} />
        </div>
      </Modal>;
  }

  render() {
    const TripListContainer = TripList.container;
    const { currentTrip, trackSegment } = this.props;
    const isConvertedTrip = isTripConverted(currentTrip.status);

    return (
      <div>
        <div className='sbc4 p8 flex alignCenter spaceBetween'>
          <div className='col-xs-7 pl0 pr4 flexFull'><TripListContainer currentTripId={currentTrip.id}
                                                                        trackSegment={trackSegment} /></div>
          <GViewPreferencesBtn className='col-xs-5 pl4 pr0'><ViewPreferencesButton currentTripId={currentTrip.id}
                                                                                   tripStatus={currentTrip.status}
                                                                                   trackSegment={trackSegment} /></GViewPreferencesBtn>
        </div>
        <div className='sbc5 pl15 pr15 pt15 pb24 flex alignCenter spaceBetween'>
          <p className='f14 pfc5 at_bookingID'>{`Booking ID #${currentTrip.id}`}</p>
          {
            isConvertedTrip
              ? null
              : this.renderCancel()
          }
        </div>
        {currentTrip.quotesCount ?
          <div className='p15 sbcw at_compareCta_section'>
            <p
              className='f14 text-center pfc3 mb5'>{`You have received ${currentTrip.quotesCount} quotes and you can`}</p>
            <a
              href={compareQuotesPath(currentTrip.id)}
              className={`text-center pt8 pb8 block wfull ${currentTrip.quotesCount > 1 ? 'btn-sec-load' : 'disabled sfcw'} at_comparequotes`}
            >
            <span className='pt2 pb2 block wfull flex alignCenter justifyCenter'>
              <span className='icon-18 mr8 iblock'>
                {
                  currentTrip.quotesCount > 1 ?  <CompareIcon /> :  <CompareIconWhite />
                }
              </span>
              <span>Compare Quotes</span>
            </span>
            </a>
          </div> : null}
      </div>
    );
  }
}

export default Header;

Header.propTypes = {
  currentTrip: PropTypes.object,
  reactivateBooking: PropTypes.func,
  trackSegment: PropTypes.func
};
