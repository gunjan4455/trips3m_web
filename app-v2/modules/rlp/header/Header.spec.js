import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Header from './Header';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  currentTrip: {
    id: '1111',
    quotesCount: 2,
    status: '',
  }
};

const shallowWrapperHeader = (props) => {
  return shallow(<Header {...props} />);
};

describe('Rlp Header: Render Correctly ', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrapperHeader(props);

    expect(wrapper).toMatchSnapshot();
  });

  it('compare quotes should be disabled for single quote', () => {
    props.currentTrip.quotesCount = 1;
    const wrapper = shallowWrapperHeader(props);

    expect(wrapper.find('.at_comparequotes').hasClass('disabled')).toEqual(true);
  });
});
