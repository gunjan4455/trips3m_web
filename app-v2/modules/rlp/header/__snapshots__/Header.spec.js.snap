// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rlp Header: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentTrip={
      Object {
        "id": "1111",
        "quotesCount": 2,
        "status": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="sbc4 p8 flex alignCenter spaceBetween"
        >
          <div
            className="col-xs-7 pl0 pr4 flexFull"
          >
            <withRouter(Connect(Dropdown))
              currentTripId="1111"
            />
          </div>
          <ForwardRef(render)
            className="col-xs-5 pl4 pr0"
          >
            <ViewPreferencesButton
              currentTripId="1111"
              tripStatus=""
            />
          </ForwardRef(render)>
        </div>,
        <div
          className="sbc5 pl15 pr15 pt15 pb24 flex alignCenter spaceBetween"
        >
          <p
            className="f14 pfc5 at_bookingID"
          >
            Booking ID #1111
          </p>
        </div>,
        <div
          className="p15 sbcw at_compareCta_section"
        >
          <p
            className="f14 text-center pfc3 mb5"
          >
            You have received 2 quotes and you can
          </p>
          <a
            className="text-center pt8 pb8 block wfull btn-sec-load at_comparequotes"
            href="/requested_trips/1111/compare"
          >
            <span
              className="pt2 pb2 block wfull flex alignCenter justifyCenter"
            >
              <span
                className="icon-18 mr8 iblock"
              >
                <IconSvg />
              </span>
              <span>
                Compare Quotes
              </span>
            </span>
          </a>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-xs-7 pl0 pr4 flexFull"
            >
              <withRouter(Connect(Dropdown))
                currentTripId="1111"
              />
            </div>,
            <ForwardRef(render)
              className="col-xs-5 pl4 pr0"
            >
              <ViewPreferencesButton
                currentTripId="1111"
                tripStatus=""
              />
            </ForwardRef(render)>,
          ],
          "className": "sbc4 p8 flex alignCenter spaceBetween",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <withRouter(Connect(Dropdown))
                currentTripId="1111"
              />,
              "className": "col-xs-7 pl0 pr4 flexFull",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "currentTripId": "1111",
                "trackSegment": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ViewPreferencesButton
                currentTripId="1111"
                tripStatus=""
              />,
              "className": "col-xs-5 pl4 pr0",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "currentTripId": "1111",
                "trackSegment": undefined,
                "tripStatus": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "div",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                Object {
                  "flex": "0 0 135px",
                },
              ],
              "defaultProps": undefined,
              "displayName": "Styled(div)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="f14 pfc5 at_bookingID"
            >
              Booking ID #1111
            </p>,
            null,
          ],
          "className": "sbc5 pl15 pr15 pt15 pb24 flex alignCenter spaceBetween",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Booking ID #1111",
              "className": "f14 pfc5 at_bookingID",
            },
            "ref": null,
            "rendered": "Booking ID #1111",
            "type": "p",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="f14 text-center pfc3 mb5"
            >
              You have received 2 quotes and you can
            </p>,
            <a
              className="text-center pt8 pb8 block wfull btn-sec-load at_comparequotes"
              href="/requested_trips/1111/compare"
            >
              <span
                className="pt2 pb2 block wfull flex alignCenter justifyCenter"
              >
                <span
                  className="icon-18 mr8 iblock"
                >
                  <IconSvg />
                </span>
                <span>
                  Compare Quotes
                </span>
              </span>
            </a>,
          ],
          "className": "p15 sbcw at_compareCta_section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You have received 2 quotes and you can",
              "className": "f14 text-center pfc3 mb5",
            },
            "ref": null,
            "rendered": "You have received 2 quotes and you can",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="pt2 pb2 block wfull flex alignCenter justifyCenter"
              >
                <span
                  className="icon-18 mr8 iblock"
                >
                  <IconSvg />
                </span>
                <span>
                  Compare Quotes
                </span>
              </span>,
              "className": "text-center pt8 pb8 block wfull btn-sec-load at_comparequotes",
              "href": "/requested_trips/1111/compare",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="icon-18 mr8 iblock"
                  >
                    <IconSvg />
                  </span>,
                  <span>
                    Compare Quotes
                  </span>,
                ],
                "className": "pt2 pb2 block wfull flex alignCenter justifyCenter",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <IconSvg />,
                    "className": "icon-18 mr8 iblock",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Compare Quotes",
                  },
                  "ref": null,
                  "rendered": "Compare Quotes",
                  "type": "span",
                },
              ],
              "type": "span",
            },
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sbc4 p8 flex alignCenter spaceBetween"
          >
            <div
              className="col-xs-7 pl0 pr4 flexFull"
            >
              <withRouter(Connect(Dropdown))
                currentTripId="1111"
              />
            </div>
            <ForwardRef(render)
              className="col-xs-5 pl4 pr0"
            >
              <ViewPreferencesButton
                currentTripId="1111"
                tripStatus=""
              />
            </ForwardRef(render)>
          </div>,
          <div
            className="sbc5 pl15 pr15 pt15 pb24 flex alignCenter spaceBetween"
          >
            <p
              className="f14 pfc5 at_bookingID"
            >
              Booking ID #1111
            </p>
          </div>,
          <div
            className="p15 sbcw at_compareCta_section"
          >
            <p
              className="f14 text-center pfc3 mb5"
            >
              You have received 2 quotes and you can
            </p>
            <a
              className="text-center pt8 pb8 block wfull btn-sec-load at_comparequotes"
              href="/requested_trips/1111/compare"
            >
              <span
                className="pt2 pb2 block wfull flex alignCenter justifyCenter"
              >
                <span
                  className="icon-18 mr8 iblock"
                >
                  <IconSvg />
                </span>
                <span>
                  Compare Quotes
                </span>
              </span>
            </a>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-xs-7 pl0 pr4 flexFull"
              >
                <withRouter(Connect(Dropdown))
                  currentTripId="1111"
                />
              </div>,
              <ForwardRef(render)
                className="col-xs-5 pl4 pr0"
              >
                <ViewPreferencesButton
                  currentTripId="1111"
                  tripStatus=""
                />
              </ForwardRef(render)>,
            ],
            "className": "sbc4 p8 flex alignCenter spaceBetween",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <withRouter(Connect(Dropdown))
                  currentTripId="1111"
                />,
                "className": "col-xs-7 pl0 pr4 flexFull",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "currentTripId": "1111",
                  "trackSegment": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ViewPreferencesButton
                  currentTripId="1111"
                  tripStatus=""
                />,
                "className": "col-xs-5 pl4 pr0",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "currentTripId": "1111",
                  "trackSegment": undefined,
                  "tripStatus": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  Object {
                    "flex": "0 0 135px",
                  },
                ],
                "defaultProps": undefined,
                "displayName": "Styled(div)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="f14 pfc5 at_bookingID"
              >
                Booking ID #1111
              </p>,
              null,
            ],
            "className": "sbc5 pl15 pr15 pt15 pb24 flex alignCenter spaceBetween",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Booking ID #1111",
                "className": "f14 pfc5 at_bookingID",
              },
              "ref": null,
              "rendered": "Booking ID #1111",
              "type": "p",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="f14 text-center pfc3 mb5"
              >
                You have received 2 quotes and you can
              </p>,
              <a
                className="text-center pt8 pb8 block wfull btn-sec-load at_comparequotes"
                href="/requested_trips/1111/compare"
              >
                <span
                  className="pt2 pb2 block wfull flex alignCenter justifyCenter"
                >
                  <span
                    className="icon-18 mr8 iblock"
                  >
                    <IconSvg />
                  </span>
                  <span>
                    Compare Quotes
                  </span>
                </span>
              </a>,
            ],
            "className": "p15 sbcw at_compareCta_section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You have received 2 quotes and you can",
                "className": "f14 text-center pfc3 mb5",
              },
              "ref": null,
              "rendered": "You have received 2 quotes and you can",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="pt2 pb2 block wfull flex alignCenter justifyCenter"
                >
                  <span
                    className="icon-18 mr8 iblock"
                  >
                    <IconSvg />
                  </span>
                  <span>
                    Compare Quotes
                  </span>
                </span>,
                "className": "text-center pt8 pb8 block wfull btn-sec-load at_comparequotes",
                "href": "/requested_trips/1111/compare",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="icon-18 mr8 iblock"
                    >
                      <IconSvg />
                    </span>,
                    <span>
                      Compare Quotes
                    </span>,
                  ],
                  "className": "pt2 pb2 block wfull flex alignCenter justifyCenter",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <IconSvg />,
                      "className": "icon-18 mr8 iblock",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Compare Quotes",
                    },
                    "ref": null,
                    "rendered": "Compare Quotes",
                    "type": "span",
                  },
                ],
                "type": "span",
              },
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
